docs/sources/flow/reference/components/remote.vault.md

```diff
@@ -0,0 +1,319 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/remote.vault
+title: remote.vault
+---
+
+# `remote.vault`
+
+`remote.vault` connect to Vault to retrieve secrets. It can retrieve secrets
+using the [KV v2][] secrets engine.
```

Comment:
Should we make a distinction and mention that it fetches a single secret that can hold multiple values?